src/lib/registration/clients/inst_scc.rb

```diff
@@ -0,0 +1,347 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2013 Novell, Inc. All Rights Reserved.
+#
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+#
+# Summary: Ask user for the SCC credentials
+#
+
+# use external rubygem for SCC communication
+require "yast/suse_connect"
+
+require "cgi"
+
+require "registration/addon"
+require "registration/exceptions"
+require "registration/helpers"
+require "registration/connect_helpers"
+require "registration/sw_mgmt"
+require "registration/storage"
+require "registration/url_helpers"
+require "registration/registration"
+require "registration/registration_ui"
+require "registration/ui/addon_eula_dialog"
+require "registration/ui/addon_selection_registration_dialog"
+require "registration/ui/addon_reg_codes_dialog"
+require "registration/ui/registered_system_dialog"
+require "registration/ui/base_system_registration_dialog"
+require "registration/ui/registration_update_dialog"
+require "registration/ui/media_addon_workflow"
+
+# TODO: move to the "Registration" name space
+module Yast
+  class InstSccClient < Client
+    include Yast::Logger
+    extend Yast::I18n
+
+    # popup message
+    CONTACTING_MESSAGE = N_("Contacting the Registration Server")
+
+    def main
+      textdomain "registration"
+      import_modules
+
+      first_run
+
+      @selected_addons = ::Registration::Storage::InstallationOptions.instance.selected_addons
+
+      initialize_regcodes
+
+      media_workflow? ? ::Registration::UI::MediaAddonWorkflow.run(WFM.Args[1]) : start_workflow
```

Comment:
this smells for me. Why not having it as two classes one for offline and one for online? with common ancestor.

Reply:
in general changing behavior when different caller call you is bad.

Reply:
OK, but that would require non trivial change and also changes outside the package. So I just added a `FIXME` to fix that later.